module03/solution.py

```diff
@@ -0,0 +1,181 @@
+import math
+
+def sum_of_digits(n):
+    """
+    Sum of all digits of a number
+    :param n: number
+    :return: sum
+    """
+    digits = to_digits(n)
+    sum = 0
+    while digits:
+        digit = digits.pop()
+        sum += digit
+
+    return sum
+
+
+def to_digits(number):
+    """
+    Split a number to decimal digits
+    :param number: an integer number
+    :return: a list of all digits of a number
+    """
+    reminder = abs(number)
+    numbers = []
+    while reminder > 0:
+        digit = reminder % 10
+        numbers.insert(0, digit)
+        reminder = reminder // 10
+
+    return numbers
+
+
+def to_number(digits):
+    """
+    Convert a list of digits to an integer number
+    :param digits: list of digits
+    :return: number from concatenated digits
+    """
+    s = ""
+    for f in digits:
+        s += str(f)
+    number = int(s)
+    return int(number)
+
+def count_vowels(str):
+    """
+    Count upper and lower case vowels in a string
+    :param str: string
+    :return: count of vowels
+    """
+    vowels = ["a", "e", "i", "o", "u", "y"]
+    s_lower = str.lower()
+    count = 0
+    for v in vowels:
+        c = s_lower.count(v)
+        count = count + c
+
+    return count
+
+
+def count_consonants(str):
+    """
+    Count upper and lower case consonants in a string
+    :param str: string
+    :return: count of consonants
+    """
+    consonants = list("bcdfghjklmnpqrstvwxz")
+    count = count_characters(str, consonants)
+
+    return count
+
+
+def count_characters(str, chars):
+    """
+    Count characters in a string
+    :param str: string
+    :param chars: list of characters to be counted
+    :return: count of all characters
+    """
+    s_lower = str.lower()
+    count = 0
+    for ch in chars:
+        c = s_lower.count(ch)
+        count = count + c
+
+    return count
+
+
+def prime_number(number):
+    """
+    Checks is given number a prime number
+    :param number: number to be checked
+    :return: True, if it's a primary or False, if it's not
+    """
+    i = 2
+    while i < number:
+        if ((number % i) == 0):
+            return False
+        i += 1
+    else:
+        return True
```

Comment:
This True does not need to be in an ELSE. If you never return False then it will come to it as the last line of the function.

Reply:
Yes, it doesn't need to be in `else`.
It's just a way to show, that `else` can be used in Python.